contrib/slurm/slurmexecutor.py

```diff
@@ -251,57 +283,120 @@ def run_slurm(benchmark, args, log_file):
-            stop()
-            return -1
+        jobid = None
+        while jobid is None:
+            with open(tmp_log, "w") as tmp_log_f:
+                srun_result = subprocess.run(
+                    srun_command,
+                    stdout=tmp_log_f,
+                    stderr=subprocess.STDOUT,
+                )
+
+            # we try to read back the log, in the first two lines there should be the jobid
+            with open(tmp_log, "r") as tmp_log_f:
+                first_lines = (
+                    tmp_log_f.readline(100).strip("\r\n")
+                    + "\\n"
+                    + tmp_log_f.readline(100).strip("\r\n")
+                )
+                logging.debug(
+                    "srun: returncode: %d, output (truncated): %s",
+                    srun_result.returncode,
+                    first_lines,
+                )
+                jobid_match = jobid_pattern.search(str(first_lines))
+                if jobid_match:
+                    jobid = int(jobid_match.group(1))
 
         seff_command = ["seff", str(jobid)]
         logging.debug(
             "Command to run: %s", " ".join(map(util.escape_string_shell, seff_command))
         )
-        result = subprocess.run(
-            seff_command,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-        )
-
-    # Runexec would populate the first 6 lines with metadata
-    with open(log_file, "r+") as file:
-        content = file.read()
-        file.seek(0, 0)
-        empty_lines = "\n" * 6
-        file.write(empty_lines + content)
 
-    exit_code, cpu_time, wall_time, memory_usage = parse_seff(str(result.stdout))
+        def get_checked_seff_result():
+            seff_result = subprocess.run(
+                seff_command,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+            if "exit code" in str(seff_result.stdout):
+                return seff_result
+            else:
+                return None
 
-    return {
-        "walltime": wall_time,
-        "cputime": cpu_time,
-        "memory": memory_usage,
-        "exitcode": ProcessExitCode.create(value=exit_code),
-    }
+        # sometimes `seff` needs a few extra seconds to realize the task has ended
+        result = wait_for(get_checked_seff_result, 30, 2)
 
+        status, exit_code, cpu_time, wall_time, memory_usage = parse_seff(
+            str(result.stdout)
+        )
 
-exit_code_pattern = re.compile(r"exit code (\d+)")
+        if os.path.exists(exitcode_file):
+            with open(exitcode_file, "r") as f:
+                returncode = int(f.read())
+                logging.debug("Exit code in file %s: %d", exitcode_file, returncode)
+        else:
+            if status == "COMPLETED":
+                raise Exception(
+                    "Should never happen: exit code not found, but task was reported COMPLETED."
+                )
```

Comment:
This indicates a bug? Then you can use `assert`.

Reply:
Sure, thanks